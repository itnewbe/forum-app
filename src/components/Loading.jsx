import React from "react";

const Loading = () => (
  <div role="progressbar" className="w-full h-1 bg-gray-200">
    <div className="h-full bg-blue-500 animate-pulse" />
  </div>
);

export default Loading;
