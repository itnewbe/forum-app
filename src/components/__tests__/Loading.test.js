import { render, screen } from "@testing-library/react";
import Loading from "../Loading";
import React from "react";

describe("Loading component", () => {
  it("should render loading bar with role progressbar", () => {
    render(<Loading />);
    const progressBar = screen.getByRole("progressbar");
    expect(progressBar).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
