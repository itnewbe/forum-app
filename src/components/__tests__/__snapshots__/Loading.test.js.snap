// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should match snapshot 1`] = `
<div>
  <div
    class="w-full h-1 bg-gray-200"
    role="progressbar"
  >
    <div
      class="h-full bg-blue-500 animate-pulse"
    />
  </div>
</div>
`;
